djstripe/models/core.py

```diff
@@ -1218,6 +1224,330 @@ def is_valid_object(cls, data):
 File = FileUpload
 
 
+class PaymentIntent(StripeModel):
+	"""
+	Stripe documentation: https://stripe.com/docs/api#payment_intents
+	"""
+
+	stripe_class = stripe.PaymentIntent
+	stripe_dashboard_item_name = "payment intents"
+
+	amount = StripeQuantumCurrencyAmountField(
+		help_text=(
+			"Amount intended to be collected by this PaymentIntent."
+		)
+	)
+	amount_capturable = StripeQuantumCurrencyAmountField(
+		help_text=(
+			"Amount that can be captured from this PaymentIntent."
+		)
+	)
+	amount_received = StripeQuantumCurrencyAmountField(
+		help_text=(
+			"Amount that was collected by this PaymentIntent."
+		)
+	)
+	# application
+	# application_fee_amount
+	canceled_at = models.DateTimeField(
+		null=True,
+		default=None,
+		help_text=(
+			"Populated when status is canceled, this is the time at which the PaymentIntent was "
+			"canceled. Measured in seconds since the Unix epoch."
+		)
+	)
+	cancellation_reason = models.CharField(
+		max_length=255,
+		null=True,
+		help_text=(
+			"User-given reason for cancellation of this PaymentIntent, one of duplicate, "
+			"fraudulent, requested_by_customer, or failed_invoice."
+		)
+	)
+	capture_method = StripeEnumField(
+		enum=enums.CaptureMethod,
+		help_text=(
+			"Capture method of this PaymentIntent, one of automatic or manual."
+		)
+	)
+	client_secret = models.CharField(
+		max_length=255,
+		help_text=(
+			"The client secret of this PaymentIntent. Used for client-side retrieval using a "
+			"publishable key."
+		)
+	)
+	confirmation_method = StripeEnumField(
+		enum=enums.ConfirmationMethod,
+		help_text=(
+			"Confirmation method of this PaymentIntent, one of manual or automatic."
+		)
+	)
+	currency = StripeCurrencyCodeField()
+	customer = models.ForeignKey(
+		"Customer",
+		null=True,
+		on_delete=models.SET_NULL,
+		help_text=(
+			"ID of the Customer this PaymentIntent is for if one exists."
+		)
+	)
+	description = models.TextField(
+		default='',
+		help_text=(
+			"An arbitrary string attached to the object. Often useful for displaying to users."
+		)
+	)
+	invoice = models.ForeignKey(
```

Comment:
I think this maybe should be a OneToOneField?

And maybe it belong on Invoice rather than here, since the PI is created from the invoice?

> The PaymentIntent associated with this invoice. The PaymentIntent is generated when the invoice is finalized, and can then be used to pay the invoice. Note that voiding an invoice will cancel the PaymentIntent.

See https://stripe.com/docs/api/invoices/object#invoice_object-payment_intent / 
https://stripe.com/docs/api/payment_intents/object#payment_intent_object-invoice

Reply:
Fixed in 7704be8c6